Main readme modifications for v0.1.0 (#598)
informalsystems/ibc-rs: README.md

# ibc-rs

[![Build Status][build-image]][build-link]
[![End to End testing][e2e-image]][e2e-link]
[![Apache 2.0 Licensed][license-image]][license-link]

> ![Rust Stable][rustc-image]

Reviewer: ```suggestion
![Rust Stable][rustc-image]
![Rust 1.49+][rustc-version]
```

Rust implementation of Interblockchain Communication (IBC) **modules** and 
**relayer**. The relayer is split into two crates: a library, and the 
relayer CLI comprising the main binary, called `hermes` (see the table below).

Includes [TLA+ specifications](/docs/spec).

> TODO: update the crate and docs links below (for relayer and relayer-cli).

| Crate name    |   Type   |     Version       | Docs |
| ------------- |:------:|:-------------:| -----:|
| [ibc (modules)](./modules) | lib|  [![IBC Crate][ibc-crate-image]][ibc-crate-link] | [![Docs][ibc-docs-image]][ibc-docs-link] |

> | [relayer](./relayer)      | lib |  [![Relayer Crate][relayer-crate-image]][relayer-crate-link]  | [![Docs][relayer-docs-image]][relayer-docs-link] |
> | [relayer-cli](./relayer-cli)  | bin: [hermes](relayer-cli/) |  [![Relayer CLI Crate][relayer-cli-crate-image]][relayer-cli-crate-link]      |  [![Docs][relayer-cli-docs-image]][relayer-cli-docs-link] |

Reviewer: ```suggestion
| [ibc-relayer](./relayer)      | lib |  [![Relayer Crate][relayer-crate-image]][relayer-crate-link]  | [![Docs][relayer-docs-image]][relayer-docs-link] |
| [ibc-relayer-cli](./relayer-cli)  | bin: [hermes](relayer-cli/) |  [![Relayer CLI Crate][relayer-cli-crate-image]][relayer-cli-crate-link]      |  [![Docs][relayer-cli-docs-image]][relayer-cli-docs-link] |
```

# ibc-rs

[![Build Status][build-image]][build-link]
[![End to End testing][e2e-image]][e2e-link]
[![Apache 2.0 Licensed][license-image]][license-link]
![Rust Stable][rustc-image]
![Rust 1.49+][rustc-version]

> Rust implementation of Interblockchain Communication (IBC) **modules** and 

Reviewer: ```suggestion
Rust implementation of Inter-Blockchain Communication (IBC) **modules** and 
```

For consistency with https://cosmos.network/ibc

> Rust implementation of Interblockchain Communication (IBC) **modules** and 
> **relayer**. The relayer is split into two crates: a library, and the 
> relayer CLI comprising the main binary, called `hermes`.
> See the table below.

Reviewer: ```suggestion
Rust implementation of the Inter-Blockchain Communication (IBC) protocol.

This project consists of the following crates:
 
 - The [`ibc`](https://docs.rs/ibc) crate defines the main data structures and on-chain logic  for the IBC protocol.
- The [`ibc-relayer`](https://docs.rs/ibc-relayer) crate provides an implementation of an IBC relayer, as a library.
- [`hermes`](https://hermes.informal.systems) is a command-line interface for the IBC relayer provided by this project.

See the table below for details.
```

[ibc-crate-image]: https://img.shields.io/crates/v/ibc.svg
[ibc-crate-link]: https://crates.io/crates/ibc
[ibc-docs-image]: https://docs.rs/ibc/badge.svg
[ibc-docs-link]: https://docs.rs/ibc/
[relayer-crate-image]: https://img.shields.io/crates/v/ibc.svg
[relayer-crate-link]: https://crates.io/crates/ibc
[relayer-docs-image]: https://docs.rs/ibc/badge.svg
[relayer-docs-link]: https://docs.rs/ibc/
[relayer-cli-crate-image]: https://img.shields.io/crates/v/ibc.svg
[relayer-cli-crate-link]: https://crates.io/crates/ibc
[relayer-cli-docs-image]: https://docs.rs/ibc/badge.svg
[relayer-cli-docs-link]: https://docs.rs/ibc/

[build-image]: https://github.com/informalsystems/ibc-rs/workflows/Rust/badge.svg
[build-link]: https://github.com/informalsystems/ibc-rs/actions?query=workflow%3ARust
[e2e-image]: https://github.com/informalsystems/ibc-rs/workflows/End%20to%20End%20testing/badge.svg
[e2e-link]: https://github.com/informalsystems/ibc-rs/actions?query=workflow%3A%22End+to+End+testing%22
[license-image]: https://img.shields.io/badge/license-Apache2.0-blue.svg
[license-link]: https://github.com/informalsystems/ibc-rs/blob/master/LICENSE

> [rustc-image]: https://img.shields.io/badge/rustc-stable-blue.svg

Reviewer: ```suggestion
[rustc-image]: https://img.shields.io/badge/rustc-stable-blue.svg
[rustc-version]: https://img.shields.io/badge/rustc-1.49+-blue.svg
```

[ibc-crate-image]: https://img.shields.io/crates/v/ibc.svg
[ibc-crate-link]: https://crates.io/crates/ibc
[ibc-docs-image]: https://docs.rs/ibc/badge.svg
[ibc-docs-link]: https://docs.rs/ibc/
[relayer-crate-image]: https://img.shields.io/crates/v/ibc.svg
[relayer-crate-link]: https://crates.io/crates/ibc
[relayer-docs-image]: https://docs.rs/ibc/badge.svg
[relayer-docs-link]: https://docs.rs/ibc/
[relayer-cli-crate-image]: https://img.shields.io/crates/v/ibc.svg
[relayer-cli-crate-link]: https://crates.io/crates/ibc
[relayer-cli-docs-image]: https://docs.rs/ibc/badge.svg
[relayer-cli-docs-link]: https://docs.rs/ibc/

[build-image]: https://github.com/informalsystems/ibc-rs/workflows/Rust/badge.svg
[build-link]: https://github.com/informalsystems/ibc-rs/actions?query=workflow%3ARust
[e2e-image]: https://github.com/informalsystems/ibc-rs/workflows/End%20to%20End%20testing/badge.svg
[e2e-link]: https://github.com/informalsystems/ibc-rs/actions?query=workflow%3A%22End+to+End+testing%22

> [license-image]: https://img.shields.io/badge/license-Apache2.0-blue.svg

Reviewer: ```suggestion
[license-image]: https://img.shields.io/badge/license-Apache_2.0-blue.svg
```
